bolt/client.go

```diff
@@ -83,39 +98,109 @@ func (c *Client) Open(ctx context.Context) error {
 		if _, err := tx.CreateBucketIfNotExists(ConfigBucket); err != nil {
 			return err
 		}
+		// Always create Build bucket.
+		if _, err := tx.CreateBucketIfNotExists(BuildBucket); err != nil {
+			return err
+		}
 		return nil
 	}); err != nil {
 		return err
 	}
 
-	// Runtime migrations
-	if err := c.OrganizationsStore.Migrate(ctx); err != nil {
-		return err
-	}
-	if err := c.SourcesStore.Migrate(ctx); err != nil {
-		return err
+	return nil
+}
+
+// Migrate moves data from an old schema to a new schema in each Store
+func (c *Client) Migrate(ctx context.Context, build chronograf.BuildInfo) error {
+	if c.db != nil {
+		// Runtime migrations
+		if err := c.OrganizationsStore.Migrate(ctx); err != nil {
+			return err
+		}
+		if err := c.SourcesStore.Migrate(ctx); err != nil {
+			return err
+		}
+		if err := c.ServersStore.Migrate(ctx); err != nil {
+			return err
+		}
+		if err := c.LayoutsStore.Migrate(ctx); err != nil {
+			return err
+		}
+		if err := c.DashboardsStore.Migrate(ctx); err != nil {
+			return err
+		}
+		if err := c.ConfigStore.Migrate(ctx); err != nil {
+			return err
+		}
+		if err := c.BuildStore.Migrate(ctx, build); err != nil {
+			return err
+		}
 	}
-	if err := c.ServersStore.Migrate(ctx); err != nil {
-		return err
+	return nil
+}
+
+// Close the connection to the bolt database
+func (c *Client) Close() error {
+	if c.db != nil {
+		return c.db.Close()
 	}
-	if err := c.LayoutsStore.Migrate(ctx); err != nil {
+	return nil
+}
+
+func (c *Client) Copy(ctx context.Context, version string) error {
+	fromFile, err := os.Open(c.Path)
+	if err != nil {
 		return err
 	}
-	if err := c.DashboardsStore.Migrate(ctx); err != nil {
+	defer fromFile.Close()
+
+	backupDir := path.Join(path.Dir(c.Path), "backup")
+	_ = os.Mkdir(backupDir, 0700)
```

Comment:
I think it would be good to Mkdir first (before the os.Open) and return its error.